pandas/tests/indexes/datetimelike.py

```diff
@@ -38,3 +38,24 @@ def test_view(self, indices):
         i_view = i.view(self._holder)
         result = self._holder(i)
         tm.assert_index_equal(result, i_view)
+
+    def test_map(self):
+        expected = self.index + 1
+        tm.assert_index_equal(self.index.map(lambda x: x + 1), expected)
+
+        series_map = pd.Series(expected, self.index)
+        tm.assert_index_equal(self.index.map(series_map), expected)
+
+        dict_map = {i: e for e, i in zip(expected, self.index)}
+        tm.assert_index_equal(self.index.map(dict_map), expected)
+
+        # map to NaT
+        result = self.index.map(lambda x: pd.NaT if x == self.index[0] else x)
+        expected = pd.Index([pd.NaT] + self.index[1:].tolist())
+        tm.assert_index_equal(result, expected)
+
+        series_map = pd.Series(expected, self.index)
+        tm.assert_index_equal(self.index.map(series_map), expected)
+
+        dict_map = {i: e for e, i in zip(expected, self.index)}
+        tm.assert_index_equal(self.index.map(dict_map), expected)
```

Comment:
One question: in the base tests you have the case of an empty mapping generating an Index of NaNs. Should there be a test for that here as well that it gives an Index of NaTs ?

Reply:
there is no easy way to do this actually, e.g. 

```
In [2]: pd.date_range('20130101', periods=3).map({})
Out[2]: Float64Index([nan, nan, nan], dtype='float64')
```

I could special case this I suppose (e.g. empty dict).

Reply:
I actually tried to fix this, but it goes down a rabbit hole. if you want to open an issue ok I guess.